hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/ReplicationSupervisor.java

```diff
@@ -232,14 +232,38 @@ private ReplicationSupervisor(StateContext context, ExecutorService executor,
    * Queue an asynchronous download of the given container.
    */
   public void addTask(AbstractReplicationTask task) {
+    if (queueHasRoomFor(task)){
+      initCounters(task);
+      addToQueue(task);
+    }
+  }
+
+  private boolean queueHasRoomFor(AbstractReplicationTask task) {
     final int max = maxQueueSize;
     if (getTotalInFlightReplications() >= max) {
       LOG.warn("Ignored {} command for container {} in Replication Supervisor"
-              + "as queue reached max size of {}.",
-          task.getClass(), task.getContainerId(), max);
-      return;
+                      + "as queue reached max size of {}.",
+              task.getClass(), task.getContainerId(), max);
+      return false;
+    }
+    return true;
+  }
+
+  private void addToQueue(AbstractReplicationTask task) {
+    if (inFlight.add(task)) {
+      if (task.getPriority() != ReplicationCommandPriority.LOW) {
+        // Low priority tasks are not included in the replication queue sizes
+        // returned to SCM in the heartbeat, so we only update the count for
+        // priorities other than low.
+        taskCounter.computeIfAbsent(task.getClass(),
+                k -> new AtomicInteger()).incrementAndGet();
+      }
+      queuedCounter.get(task.getMetricName()).incrementAndGet();
+      executor.execute(new TaskRunner(task));
     }
+  }
```

Comment:
Please move this after `initCounters` (as mentioned [previously](https://github.com/apache/ozone/pull/7776#discussion_r1934228059)).